packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -287,12 +287,21 @@ export class DeviceSession implements Disposable {
   }
 
   public async changeDeviceSettings(settings: DeviceSettings): Promise<boolean> {
+    const didOnlyReplaySettingsChanged =
+      this.deviceSettings?.replaysEnabled !== settings.replaysEnabled;
```

Comment:
you don't check if field `replaysEnabled` is the **only** difference, there might be other

if this is what you want to achieve, then please rename the reference to `didReplaySettingsChanged` or something like that